build_tools/cmake/iree_cross_compile.cmake

```diff
@@ -0,0 +1,206 @@
+  iree_to_bool(config_build_debugger "${IREE_${config_name}_BUILD_DEBUGGER}")
+  iree_to_bool(config_build_python_bindings "${IREE_${config_name}_BUILD_PYTHON_BINDINGS}")
+  iree_to_bool(config_build_experimental "${IREE_${config_name}_BUILD_EXPERIMENTAL}")
+
+  # Escape semicolons in the targets list so that CMake doesn't expand them to
+  # spaces.
+  string(REPLACE ";" "$<SEMICOLON>" config_hal_drivers_to_build "${IREE_HAL_DRIVERS_TO_BUILD}")
+  string(REPLACE ";" "$<SEMICOLON>" config_compiler_targets_to_build "${IREE_TARGET_BACKENDS_TO_BUILD}")
+  # LINT.ThenChange(https://github.com/google/iree/tree/master/CMakeLists.txt:iree_options)
+
+  message(STATUS "C compiler for ${config_name} build: ${config_c_compiler}")
+  message(STATUS "C++ compiler for ${config_name} build: ${config_cxx_compiler}")
+
+  add_custom_command(OUTPUT ${IREE_${config_name}_BINARY_ROOT}/CMakeCache.txt
+    COMMAND "${CMAKE_COMMAND}" "${PROJECT_SOURCE_DIR}" -G "${CMAKE_GENERATOR}"
+        -DCMAKE_MAKE_PROGRAM="${CMAKE_MAKE_PROGRAM}"
+        -DCMAKE_BUILD_TYPE="${CMAKE_BUILD_TYPE}"
+        -DCMAKE_C_COMPILER="${config_c_compiler}"
+        -DCMAKE_CXX_COMPILER="${config_cxx_compiler}"
+        -DIREE_ENABLE_RUNTIME_TRACING=${config_enable_runtime_tracing}
+        -DIREE_ENABLE_LLVM=${config_enable_llvm}
+        -DIREE_ENABLE_EMITC=${config_enable_emitc}
+        -DIREE_BUILD_COMPILER=${config_build_compiler}
+        -DIREE_BUILD_TESTS=${config_build_tests}
+        -DIREE_BUILD_DOCS=${config_build_docs}
+        -DIREE_BUILD_SAMPLES=${config_build_samples}
+        -DIREE_BUILD_DEBUGGER=${config_build_debugger}
+        -DIREE_BUILD_PYTHON_BINDINGS=${config_build_python_bindings}
+        -DIREE_BUILD_EXPERIMENTAL=${config_build_experimental}
+        -DIREE_HAL_DRIVERS_TO_BUILD="${config_hal_drivers_to_build}"
+        -DIREE_TARGET_BACKENDS_TO_BUILD="${config_compiler_targets_to_build}"
+    WORKING_DIRECTORY ${config_binary_root}
+    DEPENDS iree_prepare_${config_name}_dir
+    COMMENT "Configuring IREE for ${config_name} build...")
+
+  add_custom_target(iree_configure_${config_name} DEPENDS ${config_binary_root}/CMakeCache.txt)
+endfunction()
+
+# iree_get_build_command
+#
+# Gets the CMake build command for the given `target`.
+#
+# Parameters:
+# - target: the target to build.
+# - bin_dir: root binary directory containing CMakeCache.txt.
+# - cmd_var: variable name for receiving the build command.
+function(iree_get_build_command target bin_dir cmd_var)
+  cmake_parse_arguments(_RULE "" "CONFIG" "" ${ARGN})
+  if(NOT _RULE_CONFIG)
+    set(_RULE_CONFIG "$<CONFIG>")
+  endif()
+  if (CMAKE_GENERATOR MATCHES "Make")
+    # Use special command for Makefiles to support parallelism.
+    set(${cmd_var} "$(MAKE)" "-C" "${bin_dir}" "${target}" PARENT_SCOPE)
+  else()
+    set(${cmd_var} "${CMAKE_COMMAND}" --build ${bin_dir} --target ${target} --config ${_RULE_CONFIG} PARENT_SCOPE)
+  endif()
+endfunction()
+
+# iree_host_install
```

Comment:
There are references to a specific config ("host") here while the `iree_create_configuration` function above is generic. Should this also be generic, or is "host" special? Does this require calling `iree_create_configuration(HOST)`? Comments here and/or above would help.